rocketpy/mathutils/function.py

```diff
@@ -2354,58 +2374,57 @@ def __truediv__(self, other):  # pylint: disable=too-many-statements
         result : Function
             A Function object which gives the result of self(x)/other(x).
         """
-        # If other is Function try...
-        try:
-            # Check if Function objects source is array or callable
-            # Check if Function objects have the same domain discretization
-            if (
-                isinstance(other.source, np.ndarray)
-                and isinstance(self.source, np.ndarray)
-                and self.__dom_dim__ == other.__dom_dim__
-                and np.array_equal(self.x_array, other.x_array)
-            ):
-                # operate on grid values
+        self_array = isinstance(self.source, np.ndarray)
```

Comment:
Shouldn't we have an attribute that defines which type of Function self is (if it is array defined or callable)? Then you don't need this check everywhere for both self and other

Reply:
Why use the Enum? Its just simpler to have two more constants in the top no?

Reply:
I prefer using the Enum as well, it provides more readability and type safety.
We all should use Enum classes more often within this project.